nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -110,10 +113,27 @@ define([
             busEventManager = BusEventManager.make({
                 bus: runtime.bus()
             }),
-            typesToFiles = options.importData,
-            workspaceInfo = options.workspaceInfo;
+            typesToFiles = setupFileData(options.importData) || cell.metadata.kbase.bulkImportCell.inputs;
 
-        let kbaseNode = null, // the DOM element used as the container for everything in this cell
+        /**
+         * If importData exists, and has the right structure, use it.
+         * //TODO decide on right structure, should we test for knowledge about each file type?
+         * If not, and there's input data in the metadata, use that.
+         * If not, and there's nothing? throw an error.
+         * @param {object} importData
+         */
+        function setupFileData(importData) {
+            if (importData && Object.keys(importData).length) {
+                return importData;
+            }
+            const metaInputs = Utils.getCellMeta(cell, 'kbase.bulkImportCell.inputs');
+            if (metaInputs && Object.keys(metaInputs).length) {
+                return metaInputs;
+            }
+            throw new Error('No files were selected to upload!');
```

Comment:
i'm a bit unclear on the logic here and what's happening on line 116. why look up cell metadata here AND have that || case on line 116 `cell.metadata.kbase.bulkImportCell.inputs`? will we ever actually retrieve that data? 

if i'm understanding correctly, if `importData` doesn't exist, we'll look for `metaInputs`, which looks to retrieve the same data as `cell.metadata.kbase.bulkImportCell.inputs`. if neither exist, then an error is thrown, meaning we still wouldn't get to that || call (i think?)